### PR TITLE
Optimize permit sync logic

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
`TemplateEngine.permitSync` is one kind of protection provided to avoid frequent retries. Such retries may lay overheads on  cache engines and affect its efficiency. 

Currently, the protection protects the whole `TemplateEngine.Sync()` function but there is no need for some operations of the func to be protected. That is, only operations that have interactions(e.g. `queryCacheStatus` in `CheckAndUpdateRuntimeStatus`) with cache engines should be rate-limited by `TemplateEngine.permitSync`. 

This PR fixes it by permitting the operations that should be permitted.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/base/syncs.go [hunk 1]
```diff
@@ -63,9 +67,12 @@ func (t *TemplateEngine) Sync(ctx cruntime.ReconcileRequestContext) (err error)
 	}
 
 	// 4. Update runtime status
-	_, err = t.Implement.CheckAndUpdateRuntimeStatus()
-	if err != nil {
-		return
+	if t.permitSync(types.NamespacedName{Name: ctx.Name, Namespace: t.Context.Namespace}) {
+		permitted = true
```
Comment (cheyang) [inline pkg/ddc/base/syncs.go:71]: why not using `if permitted`?

Comment (TrafalgarZZZ) [reply pkg/ddc/base/syncs.go]: Thx for the suggestion. Done

**Review by cheyang — APPROVED** [review]

**Pull request merged**